debezium-core/src/main/java/io/debezium/pipeline/metrics/ChangeEventSourceMetrics.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.pipeline.metrics;
+
+import org.slf4j.Logger;
+
+import io.debezium.pipeline.source.spi.DataChangeEventListener;
+
+/**
+ * Common API for all change event source metrics regardless of the connector phase.
+ */
+interface ChangeEventSourceMetrics extends DataChangeEventListener {
+
+    void register(Logger logger);
```

Comment:
Is their any reason for not having each class set up its own local logger locally? I know we've done this kind of passing of loggers historically, but it was never quite clear to me what's the benefit of it. So unless we can identify one, I'd suggest to keep things simple(r).

Reply:
In my understanding, this method isn't about setting up a logger. It's about using a provided logger while registering the metric with an `MBeanServer` instance.

The actual registration logic is implemented in the base Metrics class: https://github.com/debezium/debezium/blob/cb0032d012aaafba9f2a57ddbdd520b7618c5b0c/debezium-core/src/main/java/io/debezium/metrics/Metrics.java#L45-L49

So if instead of accepting a logger here we want to make the implementation use its own logger, the log messages will be attributed to the abstract `Metrics` class rather than the specific metric. I don't believe this is desired.

A bigger issue to me here is the one I've stumbled upon multiple times during the development: if a metric fails to register itself (e.g. due to a logical error in the code), the connector will keep running without exposing some or any metrics which leads to less than trivial to debug integration test failures. So I'd rather say that a failure to register or unregister a metric should result in an exception logged somewhere higher in the call stack. So the logger here isn't really needed.

Reply:
> I don't believe this is desired.

That's the thing, I don't seem much value in attributing these messages to another class than `Metrics`. But we can go with this for now.

> A bigger issue to me here is...

Ok, could you log this in Jira please?

Reply:
> That's the thing

Makes sense, actually. I'll keep this in mind if I get to touch this code again.

> Ok, could you log this in Jira please?

Done: [DBZ-4541](https://issues.redhat.com/browse/DBZ-4541).